src/ome_arrow/tensor.py

```
@@ -0,0 +1,842 @@
+    if list(layout) == current:
+        return arr
+    axes = [current.index(dim) for dim in layout]
+    return np.transpose(arr, axes)
+
+
+def _dtype_from_meta(dtype: Any) -> np.dtype:
+    if dtype is None:
+        return np.dtype(np.uint16)
+    try:
+        return np.dtype(dtype)
+    except Exception:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)